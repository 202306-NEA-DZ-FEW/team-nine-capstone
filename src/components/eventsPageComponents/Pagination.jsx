import React from "react";
import { BiSolidLeftArrow, BiSolidRightArrow } from "react-icons/bi";
import ReactPaginate from "react-paginate";

const Pagination = ({ handlePageClick, pageCount }) => {
    return (
        <>
            <ReactPaginate
                breakLabel='...'
                nextLabel={
                    <span className='w-8 h-8 mx-2 hover:text-amber-400 flex items-center justify-center rounded-lg bg-gray-400'>
                        <BiSolidRightArrow />
                    </span>
                }
                onPageChange={handlePageClick}
                pageRangeDisplayed={5}
                pageCount={pageCount}
                previousLabel={
                    <span className='w-8 h-8 mx-2  hover:text-amber-400 flex items-center justify-center rounded-lg bg-gray-400'>
                        <BiSolidLeftArrow />
                    </span>
                }
                renderOnZeroPageCount={null}
                containerClassName='flex items-center justify-center my-4'
                pageClassName='block mx-2 text-lg hover:bg-gray-400 flex items-center justify-center rounded-lg w-8 h-8'
                activeClassName='bg-amber-400 text-white'
            />
        </>
    );
};

export default Pagination;
