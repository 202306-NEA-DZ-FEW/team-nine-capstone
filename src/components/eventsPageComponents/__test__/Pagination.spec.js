import renderer from "react-test-renderer";

import Pagination from "../Pagination";

describe("Pagination Snapshot", () => {
    it("should matches DOM Snapshot", () => {
        const tree = renderer.create(<Pagination />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
