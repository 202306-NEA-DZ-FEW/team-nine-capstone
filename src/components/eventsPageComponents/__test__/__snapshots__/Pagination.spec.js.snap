// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Snapshot should matches DOM Snapshot 1`] = `
<ul
  aria-label="Pagination"
  className="flex items-center justify-center my-4"
  role="navigation"
>
  <li
    className="previous disabled"
  >
    <a
      aria-disabled="true"
      aria-label="Previous page"
      className=" "
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="prev"
      role="button"
      tabIndex="-1"
    >
      <span
        className="w-8 h-8 mx-2  hover:text-amber-400 flex items-center justify-center rounded-lg bg-gray-400"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m4.431 12.822 13 9A1 1 0 0 0 19 21V3a1 1 0 0 0-1.569-.823l-13 9a1.003 1.003 0 0 0 0 1.645z"
          />
        </svg>
      </span>
    </a>
  </li>
  <li
    className="next"
  >
    <a
      aria-disabled="false"
      aria-label="Next page"
      className=""
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="next"
      role="button"
      tabIndex="0"
    >
      <span
        className="w-8 h-8 mx-2 hover:text-amber-400 flex items-center justify-center rounded-lg bg-gray-400"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A1 1 0 0 0 5 3v18a1 1 0 0 0 .536.886z"
          />
        </svg>
      </span>
    </a>
  </li>
</ul>
`;
